src/main/java/uk/gov/hmcts/sptribs/common/event/page/ApplicantDetails.java

```diff
@@ -41,7 +42,9 @@ public AboutToStartOrSubmitResponse<CaseData, State> midEvent(
         final CaseData data = details.getData();
         final List<String> errors = new ArrayList<>();
 
-        if (null != data.getCicCase() && null != data.getCicCase().getApplicantAddress()) {
+        if (null != data.getCicCase()
+            && data.getCicCase().getApplicantContactDetailsPreference().equals(ContactPreferenceType.POST)
+            && null != data.getCicCase().getApplicantAddress()) {
```

Comment:
.equals defers to ==, I believe, but == is null-safe and it also checks type compatibility at compile time.
I'd also put the entity you know exists on the LHS, so

`ContactPreferenceType.POST == data.getCicCase().getApplicantContatDetailsPreference()` ?